### PR TITLE
Nv 2641 workflows editor UI

### PR DESCRIPTION
**Pull request by BiswaViraj**
### What change does this PR introduce?

- updated the workflow editor UI
- updated the sidebar of channels and actions
- updated the workflow nodes sizes
- added new icons for each channels and actions

### Other information (Screenshots)

https://github.com/novuhq/novu/assets/29251776/203bca8c-a329-4024-bfd3-fba34b5bf43c

### CODE REVIEW

File: apps/web/src/components/layout/components/SideNav.tsx
Comment (BiswaViraj) [inline apps/web/src/components/layout/components/SideNav.tsx:1]: just re-arranged the imports and removed the unused ones

File: apps/web/src/components/layout/constants.ts [hunk 1]
```diff
@@ -1 +1 @@
-export const HEADER_HEIGHT = '40px';
+export const HEADER_HEIGHT = 40;
```
Comment (BiswaViraj) [inline apps/web/src/components/layout/constants.ts:1]: converted to num for easy addition/substraction with other height values in layouts while using `calc`

File: apps/web/src/design-system/template-button/DragButton.tsx [hunk 2]
```diff
@@ -47,57 +43,38 @@ export function DragButton({ description, label, Icon }: IDragButtonProps) {
...
-            <IconWrapper className={classes.linkIcon}>
-              <Icon />
-            </IconWrapper>
-            <StyledContentWrapper>
-              <Text weight="bold">{label}</Text>
-              <When truthy={description}>
-                <Text mt={3} color={colors.B60}>
-                  {description}
-                </Text>
-              </When>
```
Comment (BiswaViraj) [inline apps/web/src/design-system/template-button/DragButton.tsx:60]: no more needed in the new design

File: apps/web/src/pages/templates/editor/TemplateEditorPage.tsx [hunk 3]
```diff
@@ -74,7 +74,7 @@ function BaseTemplateEditorPage() {
           name="template-form"
           noValidate
           onSubmit={handleSubmit(onSubmitHandler, onInvalid)}
-          style={{ minHeight: '100%' }}
+          style={{ minHeight: '100%', height: '100%' }}
```
Comment (BiswaViraj) [inline apps/web/src/pages/templates/editor/TemplateEditorPage.tsx:77]: without this, the child div wont take the full height of form

File: apps/web/src/pages/templates/workflow/WorkflowEditor.tsx [hunk 4]
```diff
@@ -184,29 +204,44 @@ const WorkflowEditor = () => {
...
+            >
+              <Group>
+                <NameInput />
+                <Group>
+                  <UpdateButton />
+                  <Button
+                    pulse={shouldPulse}
+                    onClick={() => {
+                      navigate(basePath + '/snippet');
+                    }}
```
Comment (BiswaViraj) [inline apps/web/src/pages/templates/workflow/WorkflowEditor.tsx:223]: moved from the `Sidebar.tsx` to here according to the new design

File: apps/web/cypress/tests/notification-editor/create-notification.spec.ts [hunk 5]
```diff
@@ -9,6 +9,7 @@ describe('Creation functionality', function () {
     cy.waitLoadTemplatePage(() => {
       cy.visit('/workflows/create');
     });
+    cy.waitForNetworkIdle(500);
```
Comment (LetItRock) [inline apps/web/cypress/tests/notification-editor/create-notification.spec.ts:12]: Why do we have to add these, I see that we didn't change anything network related in this PR?

**Review by LetItRock — COMMENTED** [review]
@BiswaViraj it's awesome that you implemented also the changes in the editor itself, but we wanted to do that in the future :P but I think it's ok, it really looks cool :) 

Few things that I've noticed:

The editor should be stretched to the max available space on the screen without creating a scrollbar and the sidebar should be always visible and centered.
Screenshot 2023-08-04 at 11 49 13



There is a small padding that I've noticed.
Screenshot 2023-08-04 at 11 33 05

File: apps/web/cypress/tests/notification-editor/create-notification.spec.ts (hunk 5 above)
Comment (BiswaViraj) [reply apps/web/cypress/tests/notification-editor/create-notification.spec.ts:12]: The test cases were randomly failing on that part

**Review by LetItRock — APPROVED** [review]

**Pull request merged**